### PR TITLE
Added Spiral Matrix

### LINKED ISSUE #1921: C/CPP Reversing a word in a string
### Description

Documentation on how reverse a word  in a string

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by yellowberard**
<hr>

## Description 📜

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
Fixes #1921

<hr>

## Type of change 📝

- [x] Doc#umentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Competitive #Programming

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by yellowberard** [comment]
@Rishabh510 Please review my PR. I submitted late because I had some issues with my github so I was not able to clone.

**Comment by yellowberard** [comment]
Hi @ravireddy07 I can you tell how to format it correctly.

**Comment by ravireddy07** [comment]
> Hi @ravireddy07 I can you tell how to format it correctly.

Hey, It's simple. Just preview your readme file, It's gives output of your documenatation. then you can see what's wrong.
Whatever the editor that you using, you can preview it. Just google, how to preview a readme file with your fav editor.

you can refer other merged doc.

maintain few points like:
* Intro
* problem_Statement
* algorithm/approach
* code_snippets with explanation
* input/output examples
* complexity
* conclusion(discuss this approach and compare other approaches....etc)

**Comment by yellowberard** [comment]
I have added time complexity, example, input, output, space complexity and whole code.

File: Competitive_Programming/C++/Matrix/Spiral_Matrix.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+<h1 align="center">Displaying a Matrix in Spiral Form </h1>
+<p> In this question we will display a matrix in spiral form by traversing through <br>
+left-> right then <br>
+top-> bottom then <br>
+left->right then  <br>
+bottom-> top and 
+so on  till we reach the center element.<br>I have given a Example for you to get a better understanding:-</p>
+<h4> 
+Given a matrix of size r\*c. Traverse the matrix in spiral form.
+</h4>
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Matrix/Spiral_Matrix.md:10]: avoid using `<br>` tags, Just an empty line would be enough

File: Competitive_Programming/C++/Matrix/Spiral_Matrix.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+<p> 
+Example <br> Input:<br> r = 4, c = 4<br> matrix[][] = {{1, 2, 3, 4}, {5,
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Matrix/Spiral_Matrix.md:13]: Highlight the required words like: input, output...etc

File: Competitive_Programming/C++/Matrix/Spiral_Matrix.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+6, 7, 8}, {9, 10, 11, 12}, {13, 14, 15,16}}<br> Output: 1 2 3 4 8 12 16
+15 14 13 9 5 6 7 11 10<br>
+
+Input: r = 3, c = 4 <br> matrix[][] = {{1, 2, 3, 4}, {5, 6, 7, 8}, {9,
+10, 11, 12}}<br> Output: 1 2 3 4 8 12 11 10 9 5 6 7<br>
+
+Time Complexity:O(r*c); Space Complexity:O(r*c);<br>
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Matrix/Spiral_Matrix.md:20]: Add this at bottom with a title

File: Competitive_Programming/C++/Matrix/Spiral_Matrix.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        Solution ob;
+        vector<int> result = ob.spirallyTraverse(matrix, r, c);
+        for (int i = 0; i < result.size(); ++i)
+                cout<<result[i]<<" ";
+        cout<<endl;
+    }
+    return (0);
+
+} 
+```
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Matrix/Spiral_Matrix.md:104]: Add formatted code(use 4-space indentation)

**Review by ravireddy07 — CHANGES_REQUESTED** [review]
Please follow below points @yellowberard 

* Avoid using `<br>` tags. Just an Empty line would be enough.
* Highlight the keywords by adding some headers, bold...etc

**Review by ravireddy07 — CHANGES_REQUESTED** [review]

**Comment by yellowberard** [comment]
Can I Make a new PR

**Comment by ravireddy07** [comment]
> Can I Make a new PR

You can update same file, no need of making new PR.

**Comment by yellowberard** [comment]
I have to refork the repository because I was facing some issue with github so I can't push more commits in this PR.

**Comment by ravireddy07** [comment]
close this and raise a new PR. @yellowberard

**Comment by yellowberard** [comment]
I have created a new PR #7257.

**Pull request closed without merging**